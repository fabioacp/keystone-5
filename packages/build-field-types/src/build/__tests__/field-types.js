// @flow
import build from '../';
import fixturez from 'fixturez';
import { snapshotDirectory } from '../../../test-utils';

const f = fixturez(__dirname);

jest.setTimeout(10000);

jest.mock('../../prompt');

test('basic field type', async () => {
  let tmpPath = f.copy('basic-field-type');

  await build(tmpPath);

  await snapshotDirectory(tmpPath);
});
