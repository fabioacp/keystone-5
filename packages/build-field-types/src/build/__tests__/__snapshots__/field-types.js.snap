// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic field type: dist/basic-field-type.cjs.dev.js 1`] = `
"'use strict';

var _join = require(\\"path\\").join;

Object.defineProperty(exports, '__esModule', {
  value: true
});
let MyCoolFieldType = {
  views: {
    Field: _join(__dirname, \\"./chunk-9ca5ae48\\")
  }
};
exports.MyCoolFieldType = MyCoolFieldType;
"
`;

exports[`basic field type: dist/basic-field-type.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./basic-field-type.cjs.prod.js\\");
} else {
  module.exports = require(\\"./basic-field-type.cjs.dev.js\\");
}
"
`;

exports[`basic field type: dist/basic-field-type.cjs.prod.js 1`] = `
"\\"use strict\\";
var _join = require(\\"path\\").join;
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
let MyCoolFieldType = {
  views: { Field: _join(__dirname, \\"./chunk-9ca5ae48\\") }
};
exports.MyCoolFieldType = MyCoolFieldType;
"
`;

exports[`basic field type: dist/basic-field-type.esm.js 1`] = `
"import { join as _join } from \\"path\\";
let MyCoolFieldType = {
  views: {
    Field: _join(__dirname, \\"./chunk-605adae5\\")
  }
};
export { MyCoolFieldType };
"
`;

exports[`basic field type: dist/chunk-9ca5ae48/chunk.cjs.dev.js 1`] = `
"'use strict';

var Field = () => {
  return 'my cool react component stuff';
};

exports.default = Field;
"
`;

exports[`basic field type: dist/chunk-9ca5ae48/chunk.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./chunk.cjs.prod.js\\");
} else {
  module.exports = require(\\"./chunk.cjs.dev.js\\");
}
"
`;

exports[`basic field type: dist/chunk-9ca5ae48/chunk.cjs.prod.js 1`] = `
"\\"use strict\\";
var Field = () => \\"my cool react component stuff\\";
exports.default = Field;
"
`;

exports[`basic field type: dist/chunk-9ca5ae48/package.json 1`] = `
"{
  \\"main\\": \\"chunk.cjs.js\\",
  \\"module\\": \\"chunk.esm.js\\"
}
"
`;

exports[`basic field type: dist/chunk-605adae5/chunk.esm.js 1`] = `
"var Field = () => {
  return 'my cool react component stuff';
};

export default Field;
"
`;

exports[`basic field type: package.json 1`] = `
"{
  \\"name\\": \\"basic-field-type\\",
  \\"version\\": \\"1.0.0\\",
  \\"main\\": \\"dist/basic-field-type.cjs.js\\",
  \\"module\\": \\"dist/basic-field-type.esm.js\\",
  \\"license\\": \\"MIT\\",
  \\"private\\": true
}
"
`;

exports[`basic field type: src/index.js 1`] = `
"import { importView } from '@keystone-alpha/build-field-types';

export let MyCoolFieldType = {
  views: {
    Field: importView('./views/Field'),
  },
};
"
`;

exports[`basic field type: src/views/Field.js 1`] = `
"export default () => {
  return 'my cool react component stuff';
};
"
`;
